js-envs-test-suite/src/test/scala/org/scalajs/jsenv/test/NodeJSTest.scala

```diff
@@ -76,4 +76,35 @@ class NodeJSTest extends TimeoutComTests {
     com.await(DefaultTimeout)
   }
 
+  @Test
+  def testConcurrentSendReceive_issue3408: Unit = {
+    for (i <- 0 until 50) {
```

Comment:
50 was enough to reliably reproduce the `NullPointerException` on each run.
I also stress-tested the fix with 500.